src/main/java/org/datadog/jenkins/plugins/datadog/model/LocalCacheCounters.java

```diff
@@ -0,0 +1,14 @@
+package org.datadog.jenkins.plugins.datadog.model;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class LocalCacheCounters {
+    private static final ConcurrentMap<String, Integer> counters = new ConcurrentHashMap<String, Integer>();
```

Comment:
Let's rename this one metrics.
Also instead of a String as a key, we could also pass the metric name. So we should create a Class that would have a JSONArray for storing tags and a String for the metric name. Then we would have to implement the `equals` and `hashCode` methods so it works with HashMap. This way we could get rid of the `gauge` method we created.